marconi/src/Marconi/Indexers.hs

```diff
@@ -111,42 +112,43 @@ datumWorker Coordinator{_barrier, _channel} path = do
               offset <- findIndex (any (\(s, _) -> s < slot)) events
               Ix.rewind offset index
 
--- | does the transaction contain a targetAddress
-isInTargetTxOut
-    :: TargetAddresses        -- ^ non empty list of target address
-    -> C.TxOut C.CtxTx era    -- ^  a cardano transaction out that contains an address
-    -> Bool
-isInTargetTxOut targetAddresses (C.TxOut address _ _ _) = case address of
-    (C.AddressInEra  (C.ShelleyAddressInEra _) addr) -> addr `elem` targetAddresses
-    _                                                -> False
-
-utxoWorker
-    :: (Utxo.UtxoIndex -> IO Utxo.UtxoIndex)    -- ^ CPS function used in the queryApi thread, needs to be non-blocking
-    -> Maybe TargetAddresses                    -- ^ Target addresses to filter for
-    -> Worker
-utxoWorker indexerCallback maybeTargetAddresses Coordinator{_barrier, _channel} path = do
-    ix <- Utxo.open path (Utxo.Depth 2160)
-    workerChannel <- atomically . dupTChan $ _channel
-    void . forkIO $ innerLoop workerChannel ix
-    pure [ChainPointAtGenesis]
+utxoWorker_
+  :: (Utxo.UtxoIndexer -> IO ())  -- ^ CPS function used in the queryApi thread, needs to be non-blocking
```

Comment:
I think a callback is not CPS: in continuation passing style the callback is called as the very last statement in the function, but here the callback is called in the middle? (Unless I'm confusing something?)